pilot/pkg/config/clusterregistry/secretcontroller.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterregistry
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	// "k8s.io/client-go/tools/clientcmd"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	mcLabel      = "config.istio.io/multiClusterConfig"
+	mcAnnotation = "config.istio.io/clusterName"
```

Comment:
Let's keep it simple - use the name of the secret as 'cluster name', we can add annotation later if needed.

Also I think we can use 'config.istio.io/multiCluster' - without config. Or even: istio:multicluster (as is common style for our labels )

Reply:
ok, will remove annotation, even though I still think add flexibility in naming secrets and change the label name.